src/modules/RootPart.rb

```diff
@@ -483,11 +482,16 @@ def RunFSCKonJFS(mount_type, device, error_message)
     # @return [String] nil on success, error description on fail
     def MountPartition(mount_point, device, mount_type, fsopts = "")
       if mount_type == ""
-        # storage-ng
-        # FIXME: knowing the fstab string for the filesystem type should be
-        # responsibility of the upcoming Y2Storage::FilesystemType (to be
-        # added as part of the "yast-storage-ng as a libstorage wrapper" change)
-        mount_type = probed.partitions.with(name: device).filesystems.first.to_s
+
+        # FIXME
+        #
+        # Note that the code below will get passed unmodified fstab entries
+        # (like 'UUID=2f61fdb9-f82a-4052-8610-1eb090b82098') for device names
+        # and typically not match anything due to this.
+        #
+        # See also the comment in #update_staging_filesystem! below.
+        #
+        mount_type = fstype_for_device(probed, device) || ""
```

Comment:
Are we sure that `Filesystems::Type#to_s` is exactly the string to use in fstab? I would not trust that. As mentioned before, I would return the object in `fstype_for_device` and would use `fstype_for_device(probed, device).fstab_string` here (yes, we will need to add that function in storage-ng).

If `#fstab_string` ends up being just a call to `#to_s`, that's fine for now, but it's a implementation detail of the `Filesystems::Type` class.